var/spack/repos/builtin/packages/vtr/package.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from os.path import join as pjoin
+
+from spack.package import *
+
+
+class Vtr(MakefilePackage):
+    """The Verilog to Routing (VTR) project provides open-source CAD tools for
+    FPGA architecture and CAD research.
+    """
+
+    homepage = "https://verilogtorouting.org/"
+    git = "https://github.com/verilog-to-routing/vtr-verilog-to-routing.git"
+
+    maintainers("ueqri")  # For maintaining this package in the Spack community.
+
+    variant("eigen3", default=True, description="Build VTR with Eigen 3 library")
+
+    variant(
+        "tbb",
+        default=True,
+        description="Build VTR with Intel Threading Building Blocks (TBB) library",
+    )
+
+    variant(
+        "python",
+        default=True,
+        description="Install Python and required Python packages within Spack "
+        + "environment (not system wide)",
+    )
```

Comment:
Python seems to be a required dependency, and this variant seems to be used to have it as an "implicit" dependency on the system. Can we drop the variant and just require Python?

Reply:
I agree that Python should be explicitly required. But should we also preserve a variant that allows users to choose whether the Python packages are installed within the Spack environment or not (e.g., using `venv` instead as mentioned [here](https://docs.verilogtorouting.org/en/latest/BUILDING/#setting-up-your-environment))?

Maybe like these:

```python
variant("py3pkgs", default=False, description="Install required Python packages within Spack")
```

In terms of `python@3.8:` `py-pip` `py-wheel` `py-setuptools`, they are required regardless of the `py3pkgs` variant.

As for the build stage, the changes are as follows:

```python
def build(self, spec, prefix):
    if self.spec.satisfies("+py3pkgs") and not self.spec.satisfies("@8.0.0"):
        # Workaround for `UnicodeEncodeError: 'ascii' codec can't encode
        # characters in position ...` from pip output (the progress bar is
        # non-ascii).
        which("bash")("-c", "pip install -r requirements.txt --progress-bar off")
    make()
```

Reply:
No, what I mean is that we shouldn't rely on users to setup Python on the system without Spack knowing. It would be an implicit dependency, which is bad. Can't people use `externals` if they want to hook in system installations?

Reply:
Got it. So regarding the Python packages listed in `requirement.txt`, I guess we should also make them explicit like `depends_on("py-prettytable")` in this package. Is that right?